pkg/virt-handler/vm.go

```diff
@@ -160,6 +174,73 @@ type VirtualMachineController struct {
 	// prevents cycling an unncessary posix thread.
 	phase1NetworkSetupCache     map[types.UID]int
 	phase1NetworkSetupCacheLock sync.Mutex
+
+	domainNotifyPipes map[string]string
+}
+
+func (d *VirtualMachineController) startDomainNotifyPipe(vmi *v1.VirtualMachineInstance) (net.Listener, error) {
+
+	res, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return nil, fmt.Errorf("failed to detect isolation for launcher pod when setting up notify pipe: %v", err)
+	}
+
+	// inject the domain-notify.sock into the VMI pod.
+	socketPath := filepath.Join(res.MountRoot(), d.virtShareDir, "domain-notify-pipe.sock")
+
+	os.RemoveAll(socketPath)
+	err = os.MkdirAll(filepath.Dir(socketPath), 0755)
+	if err != nil {
+		log.Log.Reason(err).Error("unable to create directory for unix socket")
+		return nil, err
+	}
+
+	listener, err := net.Listen("unix", socketPath)
+	if err != nil {
+		log.Log.Reason(err).Error("failed to create unix socket for proxy service")
+		return nil, err
+	}
+
+	// pipe the VMI domain-notify.sock to the virt-handler domain-notify.sock
```

Comment:
This piping allows us to make this work for old and new vmis? Would it be an option to instead introduce multiple sources for the domain listener where one is the old socket and then one extra for each new VMI exists? I am not sure if this copying is necessary.

Reply:
realistically, this would mean starting a grpc server for every vmi. The way the code is structured with this notify server existing within the informer would make this pretty tedious to book keep. The pipe lets us maintain the existing informer behavior which is already pretty complex as is.

Reply:
Looking forward we should in my opinion converge everything to only one socket per virt-launcher which is used for bi-directional communication. That has for instance the advantage that we can clearly see from which launcher we get which message. Since we also already have one GRPC connection open to every virt-handler it would also not mean one more extra grpc connection.